core/capabilities/targets/write_target.go

```diff
@@ -131,16 +148,31 @@ func (cap *WriteTarget) Execute(ctx context.Context, request capabilities.Capabi
 		WorkflowExecutionID: rawExecutionID,
 		ReportId:            inputs.ID,
 	}
-	var transmitter common.Address
-	if err = cap.cr.GetLatestValue(ctx, "forwarder", "getTransmitter", primitives.Unconfirmed, queryInputs, &transmitter); err != nil {
-		return nil, fmt.Errorf("failed to getTransmitter latest value: %w", err)
+	var transmissionInfo TransmissionInfo
+	if err = cap.cr.GetLatestValue(ctx, "forwarder", "getTransmissionInfo", primitives.Unconfirmed, queryInputs, &transmissionInfo); err != nil {
+		return nil, fmt.Errorf("failed to getTransmissionInfo latest value: %w", err)
 	}
-	if transmitter != common.HexToAddress("0x0") {
-		cap.lggr.Infow("WriteTarget report already onchain - returning without a tranmission attempt", "executionID", request.Metadata.WorkflowExecutionID)
+
+	switch {
+	case transmissionInfo.State == 0: // NOT_ATTEMPTED
+		cap.lggr.Infow("non-empty report - tranasmission not attempted - attempting to push to txmgr", "request", request, "reportLen", len(inputs.Report), "reportContextLen", len(inputs.Context), "nSignatures", len(inputs.Signatures), "executionID", request.Metadata.WorkflowExecutionID)
+	case transmissionInfo.State == 1: // SUCCEEDED
+		cap.lggr.Infow("returning without a tranmission attempt - report already onchain ", "executionID", request.Metadata.WorkflowExecutionID)
+		return success(), nil
+	case transmissionInfo.State == 2: // INVALID_RECEIVER
+		cap.lggr.Infow("returning without a tranmission attempt - transmission already attempted, receiver was marked as invalid", "executionID", request.Metadata.WorkflowExecutionID)
 		return success(), nil
+	case transmissionInfo.State == 3: // FAILED
+		if transmissionInfo.GasLimit.Uint64() > cap.receiverGasMinimum {
+			cap.lggr.Infow("returning without a tranmission attempt - transmission already attempted and failed, sufficient gas was provided", "executionID", request.Metadata.WorkflowExecutionID, "receiverGasMinimum", cap.receiverGasMinimum, "transmissionGasLimit", transmissionInfo.GasLimit)
+			return success(), nil
+		} else {
+			cap.lggr.Infow("non-empty report - retrying a failed transmission - attempting to push to txmgr", "request", request, "reportLen", len(inputs.Report), "reportContextLen", len(inputs.Context), "nSignatures", len(inputs.Signatures), "executionID", request.Metadata.WorkflowExecutionID, "receiverGasMinimum", cap.receiverGasMinimum, "transmissionGasLimit", transmissionInfo.GasLimit)
```

Comment:
When will this happen? Will subsequent nodes in the transmission schedule be smart enough to bump gas limit in the future?

Reply:
This would happen if the malicious node sent less gas than required. We don't need to worry about gas bumping in the happy path.